static/src/javascripts/projects/common/modules/commercial/commercial-feature-policies.js

```diff
@@ -68,16 +68,15 @@ define([
         }
     };
 
-    policies.membershipMessages = function () {
-        if (!detect.adblockInUseSync() &&
-            detect.getBreakpoint() !== 'mobile' &&
-            config.page.contentType === 'Article' &&
-            !userFeatures.isPayingMember()
-        ) {
-            return {
-                membershipMessages : true
-            };
-        }
+    policies.checkWeCanShowMembershipMessages = function () {
```

Comment:
I'm not sure this is aligned with the way the feature policies are supposed to work. Granted, the way they operate is opaque, and that is my fault - I wrote it and, in retrospect, overengineered it.

The idea is that a policy is maps a condition or a 'mode' - 'ad free web', 'sensitive content', etc. - to a feature change - 'ad free web disables foo, bar, baz, bam...'. So a policy is named after an assertion and returns a map of booleans to show its changes. The maps of each policy then get flattened together, assuming that they contain bools.

This code will work, but it's clear that the commercial feature policies module isn't fit for purpose. The reduction function is now misleading.

For the moment, I don't think this PR needs to refactor the features module, but I would rename the returned flag after a feature 'membershipMessages' rather than a function.